implementation/src/main/java/io/smallrye/config/ConfigMappingLoader.java

```diff
@@ -35,12 +35,12 @@ public static List<ConfigMappingMetadata> getConfigMappingsMetadata(Class<?> typ
                 mappings.addAll(superType.getNested());
             }
         }
-        final ConfigMappingClass configMappingClass = ConfigMappingClass.getConfigurationClass(type);
+        ConfigMappingClass configMappingClass = ConfigMappingClass.getConfigurationClass(type);
         if (configMappingClass != null) {
             mappings.add(configMappingClass);
             mappings.addAll(getConfigMappingsMetadata(getConfigMapping(type).getInterfaceType()));
         }
-        return mappings;
+        return new ArrayList<>(mappings);
```

Comment:
Maybe this should be `List.copyOf(mappings)` instead so we don't return a mutable list.